test/e2e/upgrade_test.go

```diff
@@ -241,3 +299,154 @@ func (tc *testContext) testWorkloadsAvailable(t *testing.T) {
 		})
 	assert.NoError(t, err)
 }
+
+// createParallelUpgradesCheckerJob deploys a Job that checks the number of nodes marked as upgrading and fails if the
+// number of parallel upgrades exceeds a threshold
+func (tc *testContext) createParallelUpgradesCheckerJob() (*batchv1.Job, error) {
+	checkerCommand := `
+		#!/bin/bash
+		set -euo pipefail
+		
+		# max number of parallel upgrades allowed, fixed to 1. Refer to controllers.MaxParallelUpgrades
+		export MAX_PARALLEL_UPGRADES=1
+
+		# loop indefinitely until count exceeded
+		while true; do
+			upgradingCount=$(oc get nodes -l kubernetes.io/os=windows  -o jsonpath='{.items[*].metadata.annotations.windowsmachineconfig\.openshift\.io/upgrading}' | wc -w)	
+			if [[ $upgradingCount -gt $MAX_PARALLEL_UPGRADES ]]; then
+			  echo "error: max upgrading count exceeded"
+			  exit 1 
+			fi
+		
+			echo "pass: upgrading count $upgradingCount/$MAX_PARALLEL_UPGRADES"
+			echo "waiting 30s for next check..."
+			sleep 30
+		done
+`
+	// command
+	command := []string{"bash", "-c", checkerCommand}
+	// create job
+	nodeSelector := map[string]string{"kubernetes.io/os": "linux"}
+	podOS := v1.PodOS{Name: v1.Linux}
+	return tc.createJob(parallelUpgradesCheckerJobName, false, tc.toolsImage,
+		parallelUpgradesCheckerJobName, command, nodeSelector, []v1.Toleration{}, nil, &podOS)
+}
+
+// setupParallelUpgradesCheckerRBAC deploys all the required RBAC resources for the parallel upgrades checker job
+func setupParallelUpgradesCheckerRBAC(c *clusterinfo.OpenShift, ctx context.Context, namespace string) error {
+	if err := createParallelUpgradesCheckerServiceAccount(c, ctx, namespace); err != nil {
+		return err
+	}
+	if err := createParallelUpgradesCheckerClusterRole(c, ctx); err != nil {
+		return err
+	}
+	if err := createParallelUpgradesCheckerClusterRoleBinding(c, ctx, namespace); err != nil {
+		return err
+	}
```

Comment:
Why do we need another set of RBAC resources when we already have [a Service account](https://github.com/openshift/windows-machine-config-operator/blob/f452bc77797f5826f7f9c2bb557824dd7cdc91d9/test/e2e/validation_test.go#L302C24-L302C42) for the tests?

Reply:
To narrow the checker job RBAC to the minimum required i.e. listing nodes. 

Do you anticipate any issues? If so, I can expand the rules in the existing service account and use it to avoid deploying a  another set of RBAC resources. 

Keep in mind the job definition will be moved to a  YAML file and deployed directly with `oc`  in the script, so it might make sense to have everything there.